xlators/cluster/afr/src/afr-self-heal-common.c

```diff
@@ -1294,16 +1294,23 @@ afr_mark_split_brain_source_sinks_by_policy(
     if (fav_child == -1) {
         gf_msg(this->name, GF_LOG_ERROR, 0, AFR_MSG_SBRAIN_FAV_CHILD_POLICY,
                "No child selected by favorite-child policy.");
-    } else if (fav_child > priv->child_count - 1) {
+    } else if (fav_child > (int)(priv->child_count - 1)) {
         gf_msg(this->name, GF_LOG_ERROR, 0, AFR_MSG_SBRAIN_FAV_CHILD_POLICY,
                "Invalid child (%d) "
                "selected by policy %s.",
                fav_child, policy_str);
+        return -1;
     } else if (fav_child >= 0) {
         time = replies[fav_child].poststat.ia_mtime;
+        if (time < 0) {
```

Comment:
I think instead of putting a check on time value we should try to figure out why ia_mtime < 0, The glfsheal you are using is old(release-7) and the latest code is completely changed so i am not sure it is a right way to fix the issue. 

@pranithk @karthik-us  Can you please share your view on the same.

Reply:
figure out why ia_mtime or ia_ctime <0 maybe not so easy, but when one of the values is negative, it will make "Segmentation fault" with strftime call, and in the latest glusterfs code, it also has not the protection for it, so I suggest the protection maybe a work-around for this issue.